hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/io/HoodieSparkFileGroupReaderBasedMergeHandle.java

```diff
@@ -95,7 +95,11 @@ public void write() {
         storage.newInstance(hoodieTable.getMetaClient().getBasePath(), readerContext.getStorageConfiguration()),
         hoodieTable.getMetaClient().getBasePath().toString(), instantTime, fileSlice,
         writeSchemaWithMetaFields, writeSchemaWithMetaFields, internalSchemaOption,
-        hoodieTable.getMetaClient(), props, 0, Long.MAX_VALUE, usePosition, false)) {
+        hoodieTable.getMetaClient(), props, 0,
+        // The compaction plan does not contain the length of the bootstrap data or skeleton
```

Comment:
avoid to use hardcode `-1` to indicate the invalid length? Maybe we move it as a public static constant.